src/Client.php

```diff
@@ -1476,4 +1499,40 @@ private function getErrorCodesFromXmlElement(SimpleXMLElement $errors): array {
         }
         return $errorCodes;
     }
+
+    /**
+     * Logs a failed request.
+     *
+     * @param string $request The XML request that failed.
+     * @param string $response The XML response that was received.
+     */
+    private function logFailedRequest(string $request, string $response) {
+        // If client doesn't have a logger, there is nothing more to do here.
+        if ($this->logger === null) {
+            return;
+        }
```

Comment:
This is unnecessary if after line 141 (in the constructor) you add: `$this->setLogger(new NullLogger());`
```suggestion
```